packages/core/src/render3/animations/util.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Timing} from './interfaces';
+
+export function balanceKeyframes(
+    element: HTMLElement, styles: {[key: string]: any},
+    previousStyles?: {[key: string]: any} | null): {[key: string]: any}[] {
+  const firstKeyframe: {[key: string]: any} = {};
+  const keyframes: {[key: string]: any}[] = [firstKeyframe, styles];
+  Object.keys(styles).forEach(prop => {
+    firstKeyframe[prop] = previousStyles && previousStyles[prop] || computeStyle(element, prop);
+  });
+  if (previousStyles) {
+    Object.keys(previousStyles).forEach(prop => {
+      keyframes.forEach(kf => { kf[prop] = kf[prop] || computeStyle(element, prop); });
+    });
+  }
+  return keyframes;
+}
+
+export function computeStyle(element: HTMLElement, prop: string): string {
+  if (!window || !window.getComputedStyle) return '';
+  const gcs = window.getComputedStyle(element);
+  let val = (gcs as any)[prop];
+  val = val || gcs.getPropertyValue(prop);
+  return val;
+}
+
+/**
+ *
+ * @param element
+ * @param cb
+ */
+export function applyReflow(element: HTMLElement, cb: (reflow: number) => any) {
+  let w = element.clientWidth + 1;
+  requestAnimationFrame(() => { cb(w); });
+}
+
+export function now(): number {
+  return window ? window.performance && window.performance.now() : Date.now();
```

Comment:
There's a bug here, should be:

```suggestion
  return window && window.performance && window.performance.now() || Date.now();
```